spec/controllers/cinema_halls_controller_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe CinemaHallsController, type: :controller do
+    describe 'GET #show' do
+        it "shows cinema_hall when valid id" do
+            cinema_hall = CinemaHall.create(name: "Kyoto", capacity: 100)
+            get :show, params: {id: cinema_hall.id}
+            expect(response.status).to eq(200)
+        end
+
+        it "responds with error when not valid id" do
+            expect{get :show, params: {id: "bad_id"}}.to raise_error(ActiveRecord::RecordNotFound)
+        end
```

Comment:
Try to cover those two tests in `context` block. If you are using `... when ...` inside `it` block it means that you can easily create `context` block without big effort. Consider this:

```ruby
context "when valid ID" do
  it "shows cinema_hall" do # => consider if this description is valid, cause I did not see that you testing inside this block that cinema_hall are shown, from my perspective you testing if response status is 200 :)
    cinema_hall = CinemaHall.create(name: "Kyoto", capacity: 100)
    get :show, params: {id: cinema_hall.id}
    expect(response.status).to eq(200)
  end
end

context "when not valid ID" do
  it "responds with error" do
    expect{get :show, params: {id: "bad_id"}}.to raise_error(ActiveRecord::RecordNotFound)
  end
end
```

Reply:
Will this test check if the cinema_hall was shown?
    
```
it "shows cinema_hall" do 
          cinema_hall = CinemaHall.create(name: "Kyoto", capacity: 100)
          get :show, params: {id: cinema_hall.id}
          expect(response.status).to eq(200)
          expect(response.body).to include("Kyoto")
end
```

Reply:
Yes, for sure! But consider that this test 
```
  expect(response.body).to include("Kyoto")
```
checks if response generally includes the word "Kyoto". Better idea would to check if there is exactly this record - something like:
```
expect(JSON.parse(response.body)).to eq( { id: cinema_hall.id, name: cinema_hall.name, capacity: cinema_hall.capacity })
```

Anyway - it looks fine, I'm just considering improvements.